proto/message_contents/xmtp_envelope.proto

```diff
@@ -69,12 +67,30 @@ message MessageHeader {
 // Message is the top level protocol element
 message V1Message {
     bytes header_bytes = 1; // encapsulates the encoded MessageHeader
+    // Ciphertext.payload MUST contain encrypted EncodedContent
+    Ciphertext ciphertext = 2; 
+}
+
+// Message V2
+
+// Message header is encoded separately as the bytes are also used
+// as associated data for authenticated encryption
+message MessageHeaderV2 {
+    uint64 created_ns = 1;
+}
+
+// Message is the top level protocol element
+message MessageV2 {
+    bytes header_bytes = 1; // encapsulates encoded MessageHeaderV2
+    // Ciphertext.payload MUST contain encrypted SignedContent
     Ciphertext ciphertext = 2;
 }
 
-// The outermost wrapper for a message in a conversation topic
+
+// Versioned Message
 message Message {
     oneof version {
         V1Message v1 = 1;
+        MessageV2 v2 = 2;
```

Comment:
A little weird that we have `V1Message` and `MessageV2`. While we are at it do we not just want to rename `V1Message` to `MessageV1` for consistency? It won't change the wire format.